api/history.go

```diff
@@ -0,0 +1,66 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	apitypes "github.com/puppetlabs/wash/api/types"
+	"github.com/puppetlabs/wash/journal"
+	log "github.com/sirupsen/logrus"
+)
+
+var historyHandler handler = func(w http.ResponseWriter, r *http.Request) *errorResponse {
+	history := journal.History()
+
+	if index := mux.Vars(r)["index"]; index == "" {
+		w.WriteHeader(http.StatusOK)
+
+		commands := make([]string, len(history))
+		for i, item := range history {
+			// TODO: send structured data instead of a pre-formatted string
+			commands[i] = item.Start.Format("2006-01-02 15:04") + "  " + item.Description
+		}
+		jsonEncoder := json.NewEncoder(w)
+		if err := jsonEncoder.Encode(&commands); err != nil {
+			log.Warnf("Unable to encode history: %v", err)
+			return unknownErrorResponse(fmt.Errorf("Could not marshal %v: %v", history, err))
+		}
+	} else {
+		idx, err := strconv.Atoi(index)
+		if err != nil || idx < 0 || idx >= len(history) {
```

Comment:
Shouldn't this be based off of PID? Idx seems like something that'd be hard to track for a user.